### DISCUSSION TITLE
Using getStaticProps disables exportPathMap is this intentional?

### DISCUSSION THREAD

**Discussion by thomaschaaf** [vercel/next.js#12524] [Help]
We are currently using an `exportPathMap` which looks roughly like this:

```
{
  "/": { page: "/content", query: { foo: 'bara' } },
  "/product/": { page: "/content", query: { foo: 'barb' } },
  "/landingpage/": { page: "/content", query: { foo: 'barc' } },
  "/landingpage/info/": { page: "/content", query: { foo: 'bard' } },
  "/order/": { page: "/order" },
  "/about-us/": { page: "/about-us" },
}
```

In the /content page we are using getInitialProps to fetch the correct information depending on the query parameter. Now we wanted to migrate to getStaticsProps. With getStaticPaths we are not able to create that without creating a lot of "empty files". Sadly using getStaticProps overwrites the exportPathMap: https://github.com/zeit/next.js/blob/canary/packages/next/build/index.ts#L734

I was wondering about the future of `exportPathMap` will it be removed soon? Or would it be worth creating a PR to get both working at the same time? Is there maybe already an initiative to do that?

**Comment by timneutkens** [comment]
I think you can achieve what you're referring to with catch-all routes already: https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes and using getStaticPaths there.

**Reply by thomaschaaf** [reply]
Sadly catch all does not for the following reasons:

1. catch all’s don’t allow deeper links like so only /a would work but /a/b would not
2. this would mean all our components would have to be in one file making

**Reply by timneutkens** [reply]
> catch all’s don’t allow deeper links like so only /a would work but /a/b would not

That's exactly what catch-all routes allow you to do, eg `pages/[...slug].js` would match `/a/b` and `/a`, this is similar to what you did with having a `pages/content.js`.

> catch all’s don’t allow deeper links like so only /a would work but /a/b would not

Based on the exportPathMap you provided that's what you already do in pages/content.js.

**Reply by thomaschaaf** [reply]
Thank you for your feedback. 

For the content pages this is true but not for the order page and many more pages.
